packages/block-editor/src/components/block-patterns-list/index.js

```diff
@@ -221,24 +215,20 @@ function BlockPatternsList(
 			aria-label={ label }
 			ref={ ref }
 		>
-			{ blockPatterns.map( ( pattern ) => {
-				const isShown = shownPatterns.includes( pattern );
-				return isShown ? (
-					<BlockPattern
-						key={ pattern.name }
-						id={ pattern.name }
-						pattern={ pattern }
-						onClick={ onClickPattern }
-						onHover={ onHover }
-						isDraggable={ isDraggable }
-						showTitle={ showTitle }
-						showTooltip={ showTitlesAsTooltip }
-						category={ category }
-					/>
-				) : (
-					<BlockPatternPlaceholder key={ pattern.name } />
-				);
-			} ) }
+			{ /* To do: async render within BlockPattern. Hasn't this already been done? */ }
```

Comment:
Yes, this was implemented differently in #60425, but only for the site editor. I'll fix this in a separate PR. We should move `Async` to `components` and use it here as well.

The list change is causing all these patterns to re-render and is causing flickering.

I actually also don't see a performance regression in the PR.

Reply:
Made #66744 to handle it before or after this merge.

Reply:
Rebased on top of that now.